texts/0006-tauri-releases-schedule.md

```diff
@@ -0,0 +1,53 @@
+- Feature Name: tauri_release_schedule
+- Start Date: 2024-6-6
+- RFC PR: [tauri-apps/rfcs#14](https://github.com/tauri-apps/rfcs/pull/14)
+- Tracking Issue:
+
+# Summary
+
+Define a release schedule for tauri main repo, including crates, NPM packages and CLIs. 
+
+# Motivation
+
+As it stands now, there is no release schedule and releases are made either whenever one of the working group feels like it (mainly Lucas and myself (Amr)),
+or when there is a need for an urgent fix.
+
+As a tauri user or contributor, there is no indication when a release might happen and if will it include a certain feature
+one might be waiting for which leads me to either try to figure how to use unreleased features directly from source or wait indefinitely.   
+
+A release schedule should make the releasing process more predictable for maintainers and users and prevents the tauri project from ever appearing stagnant.
+
+# Explanation
+
+We follow semver versioning for all tauri crates, NPM packages and CLIs and so we have 6 type of releases we are ever gonna make for stable releases,
+here is what each release type cycle would look like:
+
+- `patch`: weekly
+  - `patch` versions can be released when needed, in the case of an urgent bug fix or security vulnerability. 
+  - PRs with `minor` bumps shall be marked as `Ready for next minor` and will not be merged until the next `minor` release cycle.
+    This policy exists to ease the release of `patch` versions without cherrypicking into a separate branch and releasing from there.    
+- `minor`: every 2 months
```

Comment:
2 months feels a little bit too long? Maybe because I'm working on beta so I feel the need of more frequent releases 😂, it might be fine after beta
I feel like this can be flexible just like the patch releases, I would like to think of this schedule more of a maximum amount time waiting between a feature is added to released, not a strict release schedule like major browsers

Reply:
While it does feel somewhat like a long interval, i feel like the fact that cargo lets us point to a specific commit easily makes this a non issue

Reply:
That does make sense, using nightly is always an option

I just simply not convinced with a strict release cycle, especially for a small team

Reply:
it does sound like a long period, but need to take security audits into considerations.

@tweidinger is 2 months okay for a security audit of minor releases? can we lower that to 1 month or would that make it harder to audit in time?

Reply:
if we aren't getting the audit done anyways, why not changing it to 6 weeks (similar to Rust's release schedule?)

the velocity will be there anyway, like we've seen minors with A LOT of features and changes (when we were actively working on v1 features)